src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -704,7 +665,7 @@ private IOperation CreateBoundConversionOperation(BoundConversion boundConversio
                 bool isChecked = conversion.IsNumeric && boundConversion.Checked;
                 ITypeSymbol type = boundConversion.Type;
                 Optional<object> constantValue = ConvertToOptional(boundConversion.ConstantValue);
-                return new LazyCSharpConversionExpression(operand, conversion, isExplicit, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit);
+                return new LazyCSharpConversionExpression(operand, conversion, isExplicit, isTryCast, isChecked, _semanticModel, syntax, type, constantValue, isImplicit || !isExplicit);
```

Comment:
>isImplicit || !isExplicit [](start = 158, length = 25)

We should probably rename locals to reflect what they mean (i.e. compiler generated and explicit cast). Otherwise, the code is confusing because it appears that we have to locals capture the same information in different form, but not always agree with each other.